### ISSUE TITLE
can't link

### ISSUE THREAD

**Issue by philiparvidsson** [boxdot/gurk-rs#282]
linking does not work for me. form the log:

```
2024-04-14T07:38:03.031748Z  INFO presage::manager: generating qrcode from provisioning link: sgnl://linkdevice?uuid=...&pub_key=...    
2024-04-14T07:38:13.067613Z ERROR panic: thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: ParseIntError { kind: InvalidDigit }': /home/runner/.cargo/git/checkouts/libsignal-service-rs-e91a0c40cab55da9/791c521/libsignal-service/src/provisioning/manager.rs:346    
2024-04-14T07:38:13.067769Z ERROR libsignal_service::websocket: SignalWebSocket: Websocket error: SignalWebSocket: end of application request stream; socket closing 
```

**Comment by philiparvidsson** [comment]
any thoughts on this? still does not work. do i need to do something in particular to be able to link? scanning the qr code from gurk with the ios app just results in gurk crashing with log entries as shown above.

**Comment by boxdot** [comment]
Could you maybe share the `pub_key` from the link? This is not private information. I am pretty sure that uuid is not failing when parsing, but this key. This would allow us to write a test case and fix the bug upstream in `libsignal-service`. 

Also which version are you using? The latest released one?

**Comment by philiparvidsson** [comment]
> Could you maybe share the `pub_key` from the link? This is not private information. I am pretty sure that uuid is not failing when parsing, but this key. This would allow us to write a test case and fix the bug upstream in `libsignal-service`.
> 
> Also which version are you using? The latest released one?

this is 0.4.0 which is the latest one in the AUR (using endeavouros here). log:

```
2024-06-25T03:03:30.308014Z  INFO presage::manager: generating qrcode from provisioning link: sgnl://linkdevice?uuid=...&pub_key=BRxcO8cB%2BvGxF8CS8BYnOTIMV33TPP7jM8TlWBvBycZh    
2024-06-25T03:03:54.897149Z ERROR panic: thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: ParseIntError { kind: InvalidDigit }': /home/runner/.cargo/git/checkouts/libsignal-service-rs-e91a0c40cab55da9/791c521/libsignal-service/src/provisioning/manager.rs:346    
2024-06-25T03:03:54.897277Z ERROR libsignal_service::websocket: SignalWebSocket: Websocket error: SignalWebSocket: end of application request stream; socket closing    
```

**Comment by boxdot** [comment]
This might be fixed already. I will try to release today a new version.

**Comment by alteriks** [comment]
I had the same problem using 4.3 version of gurk. After compiling git version problem went away.

**Comment by boxdot** [comment]
Just released v0.5. This should solve this issue.

**Comment by philiparvidsson** [comment]
it works now, thank you so much
